packages/blob-storage/internal/logic/callbacks.go

```diff
@@ -0,0 +1,185 @@
+	"github.com/taikoxyz/taiko-mono/packages/blob-storage/internal/taikol1"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+type Response struct {
+	Data []struct {
+		Index            string `json:"index"`
+		Blob             string `json:"blob"`
+		KzgCommitment    string `json:"kzg_commitment"`
+		KzgCommitmentHex []byte `json:"-"`
+	} `json:"data"`
+}
+
+// Callback functions
+
+// BlockProposedCallback is a callback for the "BlockProposed" event.
+func BlockProposedCallback(rpcURL, beaconURL, networkName string, log types.Log) {
+
+	contractAbi, err := abi.JSON(strings.NewReader(taikol1.TaikoL1ABI))
+	if err != nil {
+		fmt.Println("Could not initiate reader")
+	}
+
+	eventData := taikol1.TaikoL1BlockProposed{}
+
+	err = contractAbi.UnpackIntoInterface(&eventData, "BlockProposed", log.Data)
+	if err != nil {
+		fmt.Println("Could not unpack log.Data")
+	}
+
+	// Some debug logs for now.
+	// fmt.Println("Block id is:")
+	// fmt.Println(log.Topics[1])
+
+	if eventData.Meta.BlobUsed {
+		// in LibPropose we assign block.height-1 to l1Height, which is the parent block.
+		l1BlobHeight := eventData.Meta.L1Height + 1
+		blobHash := hex.EncodeToString(eventData.Meta.BlobHash[:])
+
+		storeBlob(rpcURL, beaconURL, networkName, strconv.Itoa(int(l1BlobHeight)), blobHash)
+	}
+}
+
+func getBlockTimestamp(rpcURL string, blockNumber *big.Int) (uint64, error) {
+	client, err := ethclient.Dial(rpcURL)
+	if err != nil {
+		return 0, err
+	}
+	defer client.Close()
+
+	block, err := client.BlockByNumber(context.Background(), blockNumber)
+	if err != nil {
+		return 0, err
+	}
+
+	return block.Time(), nil
+}
+
+func calculateBlobHash(commitmentStr string) string {
+	// As per: https://eips.ethereum.org/EIPS/eip-4844
+	commitment := kzg4844.Commitment(common.FromHex(commitmentStr))
+	blobHash := kzg4844.CalcBlobHashV1(
+		sha256.New(),
+		&commitment)
+	blobHashString := hex.EncodeToString(blobHash[:])
+	return blobHashString
+}
+
+func storeBlob(rpcURL, beaconURL, networkName, blockID, blobHashInMeta string) error {
+	cfg, err := GetConfig()
+	if err != nil {
+		log.Fatal("Error loading config:", err)
+	}
+	// url := fmt.Sprintf("https://l1beacon.internal.taiko.xyz/eth/v1/beacon/blob_sidecars/%s", blockID)
+	url := fmt.Sprintf("%s/%s", beaconURL, blockID)
+	response, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return err
+	}
+
+	var responseData Response
+	if err := json.Unmarshal(body, &responseData); err != nil {
+		return err
+	}
+
+	for _, data := range responseData.Data {
+
+		data.KzgCommitmentHex, err = hex.DecodeString(data.KzgCommitment[2:])
+		if err != nil {
+			return err
+		}
+
+		// Comparing the hex strings of meta.blobHash (blobHash)
+		if calculateBlobHash(data.KzgCommitment) == blobHashInMeta {
+			n := new(big.Int)
+
+			blockNrBig, ok := n.SetString(blockID, 10)
+			if !ok {
+				fmt.Println("SetString: error")
```

Comment:
Please use `log.Println`.

Reply:
specifically either use the now go standard "slog" package or the go-ethereum "log" package which implements structured logging